lightly_studio/src/lightly_studio/core/dataset.py

```diff
@@ -229,340 +206,6 @@ def __getitem__(self, key: _SliceType) -> DatasetQuery[T]:
-            session=self.session,
-            dataset_id=self.dataset_id,
-            input_labels=label_input,
-            images_path=images_path,
-        )
-
-        # Tag samples with split name if provided
-        if split is not None and created_sample_ids:
-            tag = tag_resolver.get_or_create_sample_tag_by_name(
-                session=self.session,
-                collection_id=self.dataset_id,
-                tag_name=split,
-            )
-            tag_resolver.add_sample_ids_to_tag_id(
-                session=self.session,
-                tag_id=tag.tag_id,
-                sample_ids=created_sample_ids,
-            )
-
-        if embed:
-            _generate_embeddings_image(
-                session=self.session, collection_id=self.dataset_id, sample_ids=created_sample_ids
-            )
-
-    def add_samples_from_coco_caption(
-        self,
-        annotations_json: PathLike,
-        images_path: PathLike,
-        split: str | None = None,
-        embed: bool = True,
-    ) -> None:
-        """Load a dataset in COCO caption format and store in DB.
-
-        Args:
-            annotations_json: Path to the COCO caption JSON file.
-            images_path: Path to the folder containing the images.
-            split: Optional split name to tag samples (e.g., 'train', 'val').
-                If provided, all samples will be tagged with this name.
-            embed: If True, generate embeddings for the newly added samples.
-        """
-        if isinstance(annotations_json, str):
-            annotations_json = Path(annotations_json)
-        annotations_json = annotations_json.absolute()
-
-        if not annotations_json.is_file() or annotations_json.suffix != ".json":
-            raise FileNotFoundError(f"COCO caption json file not found: '{annotations_json}'")
-
-        if isinstance(images_path, str):
-            images_path = Path(images_path)
-        images_path = images_path.absolute()
-
-        created_sample_ids = add_samples.load_into_dataset_from_coco_captions(
-            session=self.session,
-            dataset_id=self.dataset_id,
-            annotations_json=annotations_json,
-            images_path=images_path,
-        )
-
-        # Tag samples with split name if provided
-        if split is not None and created_sample_ids:
-            tag = tag_resolver.get_or_create_sample_tag_by_name(
-                session=self.session,
-                collection_id=self.dataset_id,
-                tag_name=split,
-            )
-            tag_resolver.add_sample_ids_to_tag_id(
-                session=self.session,
-                tag_id=tag.tag_id,
-                sample_ids=created_sample_ids,
-            )
-
-        if embed:
-            _generate_embeddings_image(
-                session=self.session, collection_id=self.dataset_id, sample_ids=created_sample_ids
-            )
-
-    def compute_typicality_metadata(
```

Comment:
Typicality and similarity computation should work for any collection with embeddings; isn't that the case?